src/components/KCopy/KCopy.vue

```diff
@@ -0,0 +1,263 @@
+<template>
+  <div class="k-copy">
+    <span
+      v-if="badge"
+      class="copy-badge-text"
+    >
+      {{ badgeText }}
+    </span>
+    <div
+      :class="[
+        'copy-container',
+        truncateStyles,
+        badgeStyles
+      ]"
+    >
+      <KTooltip
+        v-if="format !== 'hidden'"
+        class="copy-text"
+        :class="[truncateContent, useMono]"
+        data-testid="copy-text"
+        :label="textTooltip"
+        placement="bottomStart"
+        position-fixed
+      >
+        <span>
+          {{ textFormat }}
+        </span>
+      </KTooltip>
+
+      <KTooltip
+        class="text-icon-wrapper"
+        :label="`${props.badgeText}` ? iconTitle : tooltipText"
+        max-width="500px"
+        placement="bottomStart"
+        position-fixed
+      >
+        <KClipboardProvider v-slot="{ copyToClipboard }">
+          <CopyIcon
+            class="text-icon"
+            data-testid="copy-to-clipboard"
+            :hide-title="!!copyTooltip || undefined"
+            role="button"
+            :size="KUI_ICON_SIZE_40"
+            tabindex="0"
+            @click.stop="copyIdToClipboard(copyToClipboard)"
+          />
+        </KClipboardProvider>
+      </KTooltip>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import type { PropType } from 'vue'
+import { computed, ref } from 'vue'
+import { CopyIcon } from '@kong/icons'
+import KClipboardProvider from '@/components/KClipboardProvider'
+import KTooltip from '@/components/KTooltip/KTooltip.vue'
+import { KUI_ICON_SIZE_40 } from '@kong/design-tokens'
+
+const props = defineProps({
+  /**
+    * Text displayed before the copyable text when
+    * `badge` is true
+    */
+  badgeText: {
+    type: String,
+    default: '',
+  },
```

Comment:
This prop name is a bit confusing; it's called `badgeText` but it doesn't appear within the badge?

![image](https://github.com/Kong/kongponents/assets/2229946/da235f2e-30ac-42ec-8cb6-128118c304a0)

Reply:
This was named based on the naming that was finalized in the Component brief.

Reply:
@jillztom was the brief finalized? IMO this is confusing naming but 🤷🏼

Reply:
Let's use `badge-label` and go forward with it now.

Reply:
Updated.